stdlib/Logging/src/ConsoleLogger.jl

```diff
@@ -0,0 +1,119 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    ConsoleLogger(stream=STDERR, min_level=Info; meta_formatter=default_metafmt, show_limited=true)
+
+Logger with formatting optimized for readability in a text console, for example
+interactive work with the Julia REPL.
+
+Log levels less than `min_level` are filtered out.
+
+Message formatting can be controlled by setting keyword arguments:
+
+* `meta_formatter` is a function which takes the log event metadata
+  `(level, _module, group, id, file, line)` and produces a prefix and suffix
+  for the log message.  The default is to prefix with the log level and add a
+  suffix containing the module, file and line location.
+* `show_limited` limits the printing of large data structures to something
+  which can fit on the screen by setting the `:limit` `IOContext` key during
+  formatting.
+"""
+struct ConsoleLogger <: AbstractLogger
+    stream::IO
+    min_level::LogLevel
+    meta_formatter
+    show_limited::Bool
+    message_limits::Dict{Any,Int}
+end
+function ConsoleLogger(stream::IO=STDERR, min_level=Info;
+                       meta_formatter=default_metafmt, show_limited=true)
+    ConsoleLogger(stream, min_level, meta_formatter,
+                  show_limited, Dict{Any,Int}())
+end
+
+shouldlog(logger::ConsoleLogger, level, _module, group, id) =
+    get(logger.message_limits, id, 1) > 0
+
+min_enabled_level(logger::ConsoleLogger) = logger.min_level
+
+# Formatting of values in key value pairs
+showvalue(io, msg) = show(io, "text/plain", msg)
+function showvalue(io, e::Tuple{Exception,Any})
+    ex,bt = e
+    showerror(io, ex, bt; backtrace = bt!=nothing)
+end
+showvalue(io, ex::Exception) = showvalue(io, (ex,catch_backtrace()))
```

Comment:
I don't think we should do this --- there's no way to know whether the result of `catch_backtrace` corresponds to the given exception. This mechanism is already pretty fragile and can cause quite a bit of confusion, e.g. #19979

Reply:
One alternative is to have people pass it explicitly everywhere:

```julia
try
    # do stuff
catch ex
    @error "Extra information about something bad" exception=ex,catch_backtrace()
end
```

This might be an acceptable syntax burden?

Reply:
On further thought, to be completely safe from the problems of #19979 the pattern would have to look like

```julia
try
    # do stuff
catch ex
    bt = catch_backtrace()
    @error "Extra information about something bad" exception=ex,bt
end
```

Hmm. This breaks the general principle that log events should be lazily generated.